### DISCUSSION TITLE
DynamicMethod  not work in base object

### DISCUSSION THREAD

**Discussion by ratku** [MarimerLLC/csla#2450] [Questions]
I has Build a Business Base Class in my project to custom business logic.

when i use  DataPortal.UpdateChild(..) Method override  Child_Update(params object[] parameters) 
this method call  DynamicMethod   in file ServiceProviderMethodCaller.cs line at line 474

object relation like this:
 actual object -- >my base object -->  csla base object

if the Method like  Child_Insert()   is writen  in my base class, than call a method in  csla class and override in actual class
the override method can't  be called right.

I has fixed by writen override method in my base class again .

is there any way to fixed it in DynamicMethod  call ?

it's use in a aspnetcore 3.1 project.

**Comment by rockfordlhotka** [comment]
What version of CSLA?

**Comment by ratku** [comment]
the problem occur in  CSLA 5.5   ,CSLA   4.9  is ok,work well

**Comment by rockfordlhotka** [comment]
CSLA 5 uses a completely different way of invoking data portal operation methods because it now supports dependency injection. So the reflection used to discover the method is quite different.

When you say dynamic method, do you mean it was created with IL generation? Are you using the data portal operation attributes (soon to be required - we're phasing out the hard-coded method names like `DataPortal_XYZ`?

Can you show an example of your dynamic method?

**Comment by ratku** [comment]
1. create a BusinessBase  between csla's base class  and access class
` public abstract partial class BusinessBase<T> : Csla.BusinessBase<T> where T : BusinessBase<T>
    {
        //if Add this Code,DynamicMethod  Work 
        //protected override void DataPortal_Insert()
        //{
        //}

        /// <summary>
        /// Add This Make a Business Object can work as Child
        /// </summary>
        protected void Child_Insert()
        {
            DataPortal_Insert();
        }
    }`

2  create a access class to override DataPortal_Insert ,the class as a business object and as a  children object too!!

3  call save method ,if the  obejct has a child  as above , will  call DataPortal.UpdateChild method to update it as child
if the child is new,call  Child_Insert method in my new  BusinessBase  driver by  DynamicMethod 
in Child_Insert ,i call DataPortal_Insert method to call real access method in my access class

4  before csla4.9,it's work fine
in csla5.5,it's throw a exception says there's no DataPortal_Insert method in Csla.BusinessBase

**Comment by rockfordlhotka** [comment]
Step one is to create a unit test that reproduces the issue. That test will go here

https://github.com/MarimerLLC/csla/blob/v5.x/Source/csla.netcore.test/DataPortal/ServiceProviderMethodCallerTests.cs

**Comment by rockfordlhotka** [comment]
I think this is basically the same question being asked in this thread: https://github.com/MarimerLLC/csla/discussions/2449
